app/src/main/java/ch/epfl/polybazaar/UI/SalesOverview.java

```diff
@@ -79,15 +83,26 @@ public void onStart() {
         // prepare top menu
         TextView favorites = findViewById(R.id.favoritesOverview);
         favorites.setOnClickListener(v -> {
-            if(checkUserLoggedIn(this)) {
-                displayFavorites(this);
-            };
+            if (checkUserLoggedIn(this)) {
+                AppUser user = getUser();
+                user.getUserData().addOnSuccessListener(authUser -> {
+                    ArrayList<String> favoritesIds = authUser.getFavorites();
+
+                    // the list of favorites of the user is empty
+                    if (favoritesIds == null || favoritesIds.isEmpty()) {
+                        displayToast(this, R.string.no_favorites, Gravity.CENTER);
+                        // we relaunch the activity with the list of favorites in the bundle
+                    } else {
+                        displaySavedListings(this, favoritesIds);
+                    }
+                });
+            }
         });
 
-        // activity is launched with a list of favorites
+        // activity is launched with a list of litelistings
         Bundle bundle = getIntent().getExtras();
-        if(bundle != null) {
-            IDList = bundle.getStringArrayList("favoritesList");
+        if (bundle != null) {
+            IDList = bundle.getStringArrayList("userSavedListings");
```

Comment:
You probably should use a constant. But don't bother if it messes with Travis, we'll take care of that it the refactoring.

Reply:
Done, replaced by constant